### PR TITLE
Converts some of mech actions into balloons

### PR DESCRIPTION
**Pull request by Red-byte3D**
## About The Pull Request
part 1 converting of mech text into ballons

## Why It's Good For The Game
ballons look better and work better
## Changelog
:cl:
qol: Converts most of the equipment on mechs into ballons
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
im making it async hold on

File: code/modules/vehicles/mecha/combat/durand.dm [hunk 1]
```diff
@@ -189,12 +189,12 @@ own integrity back to max. Shield is automatically dropped if we run out of powe
 	if(switching && !signal_args[1])
 		return
 	if(!chassis.defense_mode && (!chassis.cell || chassis.cell.charge < 100)) //If it's off, and we have less than 100 units of power
-		to_chat(currentuser, "[icon2html(src, currentuser)]<span class='warn'>Insufficient power; cannot activate defense mode.</span>")
+		chassis.balloon_alert(owner, "Insufficient power; cannot activate defense mode.")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/combat/durand.dm:192]:
```suggestion
		chassis.balloon_alert(owner, "insufficient power")
```

1. Arm has made them lowercase, and I've decided I like that.
2. We already know we're activating the shield, so it's unnecessary to state it again. Normally it wouldn't matter, but balloon alrets are best kept short.

File: code/modules/vehicles/mecha/combat/durand.dm [hunk 2]
```diff
@@ -189,12 +189,12 @@ own integrity back to max. Shield is automatically dropped if we run out of powe
...
 		return
 	switching = TRUE
 	chassis.defense_mode = !chassis.defense_mode
 	if(!signal_args[1])
-		to_chat(currentuser, "[icon2html(src, currentuser)]<span class='notice'>Defense mode [chassis.defense_mode?"enabled":"disabled"].</span>")
+		chassis.balloon_alert(owner, "Defense mode [chassis.defense_mode?"enabled":"disabled"]")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/combat/durand.dm:197]:
```suggestion
		chassis.balloon_alert(owner, "defense mode [chassis.defense_mode?"enabled":"disabled"]")
```

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 3]
```diff
@@ -35,7 +35,7 @@
 		return
 	chassis.use_internal_tank = !chassis.use_internal_tank
 	button_icon_state = "mech_internals_[chassis.use_internal_tank ? "on" : "off"]"
-	to_chat(chassis.occupants, "[icon2html(chassis, owner)]<span class='notice'>Now taking air from [chassis.use_internal_tank?"internal airtank":"environment"].</span>")
+	chassis.balloon_alert(owner, "Now taking air from [chassis.use_internal_tank ? "internal airtank" : "environment"]")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:38]:
```suggestion
	chassis.balloon_alert(owner, "taking air from [chassis.use_internal_tank ? "internal tank" : "environment"]")
```

Short, succinct, lowercase.

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 4]
```diff
@@ -54,11 +54,11 @@
 			available_equipment += equipment
 
 	if(available_equipment.len == 0)
-		to_chat(owner, "[icon2html(chassis, owner)]<span class='warning'>No equipment available!</span>")
+		chassis.balloon_alert(owner, "No Equipment Available")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:57]:
```suggestion
		chassis.balloon_alert(owner, "no equipment available")
```

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 5]
```diff
@@ -54,11 +54,11 @@
...
 		return
 	if(!chassis.selected)
 		chassis.selected = available_equipment[1]
-		to_chat(owner, "[icon2html(chassis, owner)]<span class='notice'>You select [chassis.selected].</span>")
+		chassis.balloon_alert(owner, "You select [chassis.selected]")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:61]:
```suggestion
		chassis.balloon_alert(owner, "[chassis.selected] selected")
```

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 6]
```diff
@@ -70,11 +70,11 @@
 			continue
 		if(available_equipment.len == number)
 			chassis.selected = null
-			to_chat(owner, "[icon2html(chassis, owner)]<span class='notice'>You switch to no equipment.</span>")
+			chassis.balloon_alert(owner, "You switch to no equipment")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:73]:
```suggestion
			chassis.balloon_alert(owner, "switched to no equipment")
```

Again, I want to keep these very short (and lowercase).

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 7]
```diff
@@ -70,11 +70,11 @@
...
 			button_icon_state = "mech_cycle_equip_off"
 		else
 			chassis.selected = available_equipment[number+1]
-			to_chat(owner, "[icon2html(chassis, owner)]<span class='notice'>You switch to [chassis.selected].</span>")
+			chassis.balloon_alert(owner, "You switch to [chassis.selected]")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:77]:
```suggestion
			chassis.balloon_alert(owner, "switched to [chassis.selected]")
```

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 8]
```diff
@@ -89,15 +89,15 @@
 	if(!owner || !chassis || !(owner in chassis.occupants))
 		return
 	if(!(chassis.mecha_flags & HAS_LIGHTS))
-		to_chat(owner, "<span class='warning'>This mechs lights are destroyed!</span>")
+		chassis.balloon_alert(owner, "This mechs lights are destroyed!")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:92]:
```suggestion
		chassis.balloon_alert(owner, "mech lights are destroyed!")
```

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 9]
```diff
@@ -89,15 +89,15 @@
...
 		return
 	chassis.mecha_flags ^= LIGHTS_ON
 	if(chassis.mecha_flags & LIGHTS_ON)
 		button_icon_state = "mech_lights_on"
 	else
 		button_icon_state = "mech_lights_off"
 	chassis.set_light_on(chassis.mecha_flags & LIGHTS_ON)
-	to_chat(owner, "[icon2html(chassis, owner)]<span class='notice'>Toggled lights [(chassis.mecha_flags & LIGHTS_ON)?"on":"off"].</span>")
+	to_chat(owner, "Toggled lights [chassis.mecha_flags & LIGHTS_ON ? "on":"off"]")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:100]:
```suggestion
	to_chat(owner, "lights [chassis.mecha_flags & LIGHTS_ON ? "on":"off"]")
```

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 10]
```diff
@@ -167,11 +167,11 @@
 	if(chassis.leg_overload_mode)
 		chassis.movedelay = min(1, round(chassis.movedelay * 0.5))
 		chassis.step_energy_drain = max(chassis.overload_step_energy_drain_min,chassis.step_energy_drain*chassis.leg_overload_coeff)
-		to_chat(owner, "[icon2html(chassis, owner)]<span class='danger'>You enable leg actuators overload.</span>")
+		chassis.balloon_alert(owner,"You enable leg actuators overload.")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:170]:
```suggestion
		chassis.balloon_alert(owner, "leg actuators overload enabled")
```

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 11]
```diff
@@ -167,11 +167,11 @@
...
 	else
 		chassis.movedelay = initial(chassis.movedelay)
 		chassis.step_energy_drain = chassis.normal_step_energy_drain
-		to_chat(owner, "[icon2html(chassis, owner)]<span class='notice'>You disable leg actuators overload.</span>")
+		chassis.balloon_alert(owner, "You disable the leg actuators overload.")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:174]:
```suggestion
		chassis.balloon_alert(owner, "leg actuators overload disabled")
```

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 12]
```diff
@@ -217,13 +217,13 @@
 	switch(chassis.damtype)
 		if("tox")
 			new_damtype = "brute"
-			to_chat(owner, "[icon2html(chassis, owner)]<span class='notice'>Your exosuit's hands form into fists.</span>")
+			chassis.balloon_alert(owner, "Your exosuit's hands form into fists.")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:220]:
```suggestion
			chassis.balloon_alert(owner, "your exosuit's hands form into fists")
```

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 13]
```diff
@@ -217,13 +217,13 @@
...
 		if("brute")
 			new_damtype = "fire"
-			to_chat(owner, "[icon2html(chassis, owner)]<span class='notice'>A torch tip extends from your exosuit's hand, glowing red.</span>")
+			chassis.balloon_alert(owner, "A torch tip extends from your exosuit's hand, glowing red.")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:223]:
```suggestion
			chassis.balloon_alert(owner, "a torch tip extends from your exosuit's hand, glowing red")
```

(I can't really find a way to shorten this, but I worry it's too long).

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 14]
```diff
@@ -238,5 +238,5 @@
 		return
 	chassis.phasing = !chassis.phasing
 	button_icon_state = "mech_phasing_[chassis.phasing ? "on" : "off"]"
-	to_chat(owner, "[icon2html(chassis, owner)]<font color=\"[chassis.phasing?"#00f\">En":"#f00\">Dis"]abled phasing.</font>")
+	chassis.balloon_alert(owner, "[chassis.phasing ? "Enabled" : "Disabled"] phasing")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:241]:
```suggestion
	chassis.balloon_alert(owner, "phasing [chassis.phasing ? "enabled" : "disabled"]")
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Very good work, thanks a lot.

I want to keep balloon alerts very short (since they're temporary), and lowercase (after seeing Arm's PR). I'm going to write a doc for this very soon, but thanks for helping me figure this out.

**Comment by tralezab** [comment]
Actually it was tralezabs idea

File: code/modules/vehicles/mecha/combat/durand.dm [hunk 15]
```diff
@@ -189,12 +189,12 @@ own integrity back to max. Shield is automatically dropped if we run out of powe
 	if(switching && !signal_args[1])
 		return
 	if(!chassis.defense_mode && (!chassis.cell || chassis.cell.charge < 100)) //If it's off, and we have less than 100 units of power
-		to_chat(currentuser, "[icon2html(src, currentuser)]<span class='warn'>Insufficient power; cannot activate defense mode.</span>")
+		chassis.balloon_alert(owner, "insufficient power.")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/combat/durand.dm:192]:
```suggestion
		chassis.balloon_alert(owner, "insufficient power")
```

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 16]
```diff
@@ -54,11 +54,11 @@
 			available_equipment += equipment
 
 	if(available_equipment.len == 0)
-		to_chat(owner, "[icon2html(chassis, owner)]<span class='warning'>No equipment available!</span>")
+		chassis.balloon_alert(owner, "no equipment available")
 		return
 	if(!chassis.selected)
 		chassis.selected = available_equipment[1]
-		to_chat(owner, "[icon2html(chassis, owner)]<span class='notice'>You select [chassis.selected].</span>")
+		chassis.balloon_alert(owner, "you select [chassis.selected]")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:61]:
```suggestion
		chassis.balloon_alert(owner, "[chassis.selected] selected")
```

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 17]
```diff
@@ -167,11 +167,11 @@
 	if(chassis.leg_overload_mode)
 		chassis.movedelay = min(1, round(chassis.movedelay * 0.5))
 		chassis.step_energy_drain = max(chassis.overload_step_energy_drain_min,chassis.step_energy_drain*chassis.leg_overload_coeff)
-		to_chat(owner, "[icon2html(chassis, owner)]<span class='danger'>You enable leg actuators overload.</span>")
+		chassis.balloon_alert(owner,"you overload the leg actuators overload.")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:170]:
```suggestion
		chassis.balloon_alert(owner,"leg actuators overloaded")
```

It's wordy.

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 18]
```diff
@@ -167,11 +167,11 @@
...
 	else
 		chassis.movedelay = initial(chassis.movedelay)
 		chassis.step_energy_drain = chassis.normal_step_energy_drain
-		to_chat(owner, "[icon2html(chassis, owner)]<span class='notice'>You disable leg actuators overload.</span>")
+		chassis.balloon_alert(owner, "you disable the overload.")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:174]:
```suggestion
		chassis.balloon_alert(owner, "you disable the overload")
```

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 19]
```diff
@@ -217,13 +217,13 @@
 	switch(chassis.damtype)
 		if("tox")
 			new_damtype = "brute"
-			to_chat(owner, "[icon2html(chassis, owner)]<span class='notice'>Your exosuit's hands form into fists.</span>")
+			chassis.balloon_alert(owner, "your punches will now do brute damage")
 		if("brute")
 			new_damtype = "fire"
-			to_chat(owner, "[icon2html(chassis, owner)]<span class='notice'>A torch tip extends from your exosuit's hand, glowing red.</span>")
+			chassis.balloon_alert(owner, "your punches will now do fire damage")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:223]:
```suggestion
			chassis.balloon_alert(owner, "your punches will now deal burn damage")
```

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 20]
```diff
@@ -217,13 +217,13 @@
 	switch(chassis.damtype)
 		if("tox")
 			new_damtype = "brute"
-			to_chat(owner, "[icon2html(chassis, owner)]<span class='notice'>Your exosuit's hands form into fists.</span>")
+			chassis.balloon_alert(owner, "your punches will now do brute damage")
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mecha_actions.dm:220]:
```suggestion
			chassis.balloon_alert(owner, "your punches will now deal brute damage")
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Closer, but you missed a few of my suggestions from earlier.

**Pull request merged**